syntax = "proto3";

package Ydb.KeyValue.V2;

option java_package = "com.yandex.ydb.keyvalue.v2";
option java_outer_classname = "KeyValueGrpc";
option java_multiple_files = true;

import "ydb/public/api/protos/ydb_keyvalue.proto";

// KeyValue tablets provide a simple key-value storage in a low-overhead and easy-to-shoot-your-leg manner.
// To use KeyValue tablets in an efficient way one must be familiar with the design of both the KeyValue tablet
// and the Distributed Storage underneath it.

service KeyValueService {

    // Acquire an exclusive lock for the partition.
    rpc AcquireLock(KeyValue.AcquireLockRequest) returns (KeyValue.AcquireLockResult);

    // Perform list of commands to modify the state of the partition as an atomic transaction.
    rpc ExecuteTransaction(KeyValue.ExecuteTransactionRequest) returns (KeyValue.ExecuteTransactionResult);

    // Read the value stored in the item with the key specified.
    rpc Read(KeyValue.ReadRequest) returns (KeyValue.ReadResult);

    // Read items with keys in the specified range.
    rpc ReadRange(KeyValue.ReadRangeRequest) returns (KeyValue.ReadRangeResult);

    // List keys and metadata of items with keys in the specified range.
    rpc ListRange(KeyValue.ListRangeRequest) returns (KeyValue.ListRangeResult);

    // Get storage channel status of the partition.
    rpc GetStorageChannelStatus(KeyValue.GetStorageChannelStatusRequest) returns (KeyValue.GetStorageChannelStatusResult);

}
